client/app/telecom/telephony/alias/configuration/changeType/telecom-telephony-alias-configuration-changeType.controller.js

```diff
@@ -77,6 +78,109 @@ angular.module("managerApp").controller("TelecomTelephonyAliasConfigurationChang
 
     /* -----  End of INITIALIZATION  ------*/
 
+    /* ===========================
+    =            BULK            =
+    ============================ */
+
+    self.bulkDatas = {
+        billingAccount: $stateParams.billingAccount,
+        serviceName: $stateParams.serviceName,
+        infos: {
+            name: "configurationNumberChangeType",
+            actions: [{
+                name: "options",
+                route: "/telephony/{billingAccount}/number/{serviceName}/changeFeatureType",
+                method: "POST",
+                params: null
+            }]
+        }
+    };
+
+    self.getBulkParams = function () {
+        var data = {
+            featureType: self.number.feature.featureType
+        };
+
+        return data;
+    };
+
+    self.onBulkSuccess = function (bulkResult) {
+
+        // check if server tasks are all successful
+        self.checkServerTasksStatus(bulkResult.success).then(function (errorCount) {
+            if (errorCount > 0) {
+                Toast.warn([$translate.instant("telephony_alias_config_change_type_bulk_server_tasks_some_error")]);
+            }
+
+            telephonyBulk.getToastInfos(bulkResult, {
+                fullSuccess: $translate.instant("telephony_alias_config_change_type_bulk_all_success"),
+                partialSuccess: $translate.instant("telephony_alias_config_change_type_bulk_some_success", {
+                    count: bulkResult.success.length
+                }),
+                error: $translate.instant("telephony_alias_config_change_type_bulk_error")
+            }).forEach(function (toastInfo) {
+                Toast[toastInfo.type](toastInfo.message, {
+                    hideAfter: null
+                });
+            });
+
+            self.changeType();
+            self.noCache = true;
+
+            // reset initial values to be able to modify again the options
+            init();
+        }, function () {
+            Toast.error([$translate.instant("telephony_alias_config_change_type_bulk_server_tasks_all_error")]);
+        });
+    };
+
+    self.onBulkError = function (error) {
+        Toast.error([$translate.instant("telephony_alias_config_change_type_bulk_on_error"), _.get(error, "msg.data")].join(" "));
+    };
+
+    self.checkServerTasksStatus = function (updatedServices) {
+        self.loading.changing = true;
+
+        var deferred = $q.defer();
+        var promises = [];
+        var tasksAndServicesInfos = [];
+
+        _.forEach(updatedServices, function (service) {
+            var info = {
+                billingAccount: service.billingAccount,
+                serviceName: service.serviceName,
+                id: _.chain(service.values).map("value").filter(function (val) {
+                    return val.action === "changeType";
+                }).value()[0].taskId
+            };
+
+            tasksAndServicesInfos.push(info);
+        });
+
+        _.forEach(tasksAndServicesInfos, function (task) {
```

Comment:
This mean you launch as many poll as there is number selected. It would be better to launch one poll for the first task returned, wait that task is done (poll in success or error) and then launch the second poll ofr the second task, etc

Reply:
Yes, i'm currently reviewing this code, it don't behave as expected.